packages/database/data-migrations/20241120150728_product_revamp/data-migration.ts

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable no-console -- logging is allowed in migration scripts */
+import { Prisma, PrismaClient } from "@prisma/client";
+
+const prisma = new PrismaClient();
+const TRANSACTION_TIMEOUT = 3 * 60 * 1000; // 3 minutes in milliseconds
+
+type Plan = "free" | "startup" | "scale" | "enterprise";
+
+const projectsLimitByPlan: Record<Plan, number | null> = {
+  free: 3,
+  startup: 3,
+  scale: 5,
+  enterprise: null,
+};
+
+async function runMigration(): Promise<void> {
+  const startTime = Date.now();
+  console.log("Starting data migration...");
+
+  await prisma.$transaction(
+    async (transactionPrisma) => {
+      const organizations = await transactionPrisma.organization.findMany({
+        where: {
+          billing: {
+            path: ["limits", "projects"],
+            equals: Prisma.DbNull,
+          },
+        },
+        select: {
+          id: true,
+          billing: true,
+        },
+      });
+
+      const updateOrganizationPromises = organizations.map((org) =>
+        transactionPrisma.organization.update({
```

Comment:
what does the data migration do on self-hosted instances where there is no billing object?

Reply:
we always have the organisation's billing object, and we just don't use it anywhere for self-hosters, so this data migration would just add the projects to the billing object without causing any error.
Should we run this data migration only for the Formbricks cloud?